### PR TITLE
adjusted the repo structure to get it ready for including azure pipeline task

### PR DESCRIPTION
**Pull request by zaowang-ms**
This is the pull request to include azure pipeline task as discussed.

### CODE REVIEW

**Comment by ornj** [comment]
This PR is a bit more than adjusting the repo structure. I recommend sticking to the structure refactor so that can get approved with little to no controversy and then opening a new PR to add the Azure Pipeline. That way we can have the appropriate people review it without getting distracted by the refactor.

@amudukutore I think we need to add doc ops and an engineer with Typescript experience (@bheftel ?) to get @zaowang-ms's work reviewed.

**Comment by zaowang-ms** [comment]
> This PR is a bit more than adjusting the repo structure. I recommend sticking to the structure refactor so that can get approved with little to no controversy and then opening a new PR to add the Azure Pipeline. That way we can have the appropriate people review it without getting distracted by the refactor.
> 
> @amudukutore I think we need to add doc ops and an engineer with Typescript experience (@bheftel ?) to get @zaowang-ms's work reviewed.

Good suggestion. Let's have a follow up discussion as Ashok mentioned and then I can adjust as what we come up with

**Comment by amudukutore** [comment]
@zaowang-ms - agreed with suggestion from @ornj on refactoring the PR into 2 separate ones - one for the restructure (which we can review between Steve and myself) and one for the rest of the Typescript (which we can request @bheftel to review).

**Comment by zaowang-ms** [comment]
> @zaowang-ms - agreed with suggestion from @ornj on refactoring the PR into 2 separate ones - one for the restructure (which we can review between Steve and myself) and one for the rest of the Typescript (which we can request @bheftel to review).

Thanks for taking a look. Good idea, I'll proceed with your suggestions and close this PR.

**Comment by zaowang-ms** [comment]
@ornj  @amudukutore  Instead of close this current PR, I adjusted the files and make this pr only for restructuring. Now the changes should be easier to review. For adding the additional azure_pipelines, created a separate PR here https://github.com/nginxinc/nginx-for-azure-deploy-action/pull/29

File: .github/workflows/testNginxForAzureDeploy.yml
[the remark this answers is not in the document]
Comment (zaowang-ms) [reply .github/workflows/testNginxForAzureDeploy.yml:1]: Good point. I think we can go with kebab-case as it's already there for many files. I have changed the file and folder names as well as the paths inside the yaml to kebab.

**Pull request merged**